python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,237 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+from __future__ import annotations
+
+import getpass
+
+import boto3
+from datetime import datetime
+from typing import Union, Optional, List, Set, Dict
+
+from pyiceberg.catalog.hive import OWNER
+
+from pyiceberg.catalog import Catalog, PropertiesUpdateSummary
+from pyiceberg.schema import Schema
+from pyiceberg.table import Table
+from pyiceberg.table.partitioning import PartitionSpec, UNPARTITIONED_PARTITION_SPEC
+from pyiceberg.table.sorting import SortOrder, UNSORTED_SORT_ORDER, SortDirection
+from pyiceberg.typedef import Identifier, Properties, EMPTY_DICT
+from pyiceberg.types import NestedField
+
+
+class GlueCatalog(Catalog):
+
+    def __init__(self, name: str, properties: Properties):
+        self.client = boto3.client("glue")
+        self.sts_client = boto3.client("sts")
+        super().__init__(name, **properties)
+
+    def _check_response(self, response: Dict[str, Dict[str, str]]):
+        if response['ResponseMetadata']['HTTPStatusCode'] != 200:
+            raise ValueError(f"Got unexpected status code {response['HttpStatusCode']}")
+
+    def _glue_to_iceberg(self, glue_table):
+        # TODO
+        pass
+
+    def _iceberg_to_glue(self, iceberg_table):
+        # TODO
+        pass
+
+    # tested on pre-existing database
+    def create_table(
```

Comment:
It can create table on pre-existing database. However, sometimes the created table cannot be viewed on GlueGatalog UI. (even though load_table and list_tables can find them). May due to invalid permission setting of the database